src/tour.ts

```diff
@@ -81,11 +80,60 @@ export interface TourOptions {
   useModalOverlay?: boolean;
 }
 
+const SHEPHERD_DEFAULT_API = 'https://shepherdpro.com';
+const SHEPHERD_USER_ID = 'shepherdPro:userId';
+
+export class ShepherdPro extends Evented {
+  public activeTour: Tour | null | undefined;
+  public apiKey: string | undefined;
+  public apiPath: string | undefined;
+  public dataRequester: DataRequest | undefined;
+
+  init(apiKey: string, apiPath?: string) {
+    if (!apiKey) {
+      throw new Error('Shepherd Pro: Missing required apiKey option.');
+    }
+    this.apiKey = apiKey;
+    this.apiPath = apiPath ?? SHEPHERD_DEFAULT_API;
+
+    if (this.apiKey) {
+      this.dataRequester = new DataRequest(this.apiKey, this.apiPath);
+      // Setup actor before first tour is loaded if none exists
+      const shepherdProId = localStorage.getItem(SHEPHERD_USER_ID);
+
+      if (!shepherdProId) {
+        this.createNewActor();
+      }
+    }
+  }
+
+  async createNewActor() {
+    if (!this.dataRequester) return;
+
+    // Setup type returns an actor
+    const response = (await this.dataRequester.sendEvents({
+      data: {
+        currentUserId: null,
+        eventType: 'setup'
+      }
+    })) as unknown as Actor;
+
+    localStorage.setItem(SHEPHERD_USER_ID, String(response.actorId));
+  }
+}
+
+const Shepherd = new ShepherdPro();
+
 /**
  * Class representing the site tour
  * @extends {Evented}
  */
 export class Tour extends Evented {
+  public dataRequester;
+  public trackedEvents = ['active', 'cancel', 'complete', 'show'];
```

Comment:
```suggestion
  dataRequester;
  trackedEvents = ['active', 'cancel', 'complete', 'show'];
```

I believe `public` is implied if it is not marked private.